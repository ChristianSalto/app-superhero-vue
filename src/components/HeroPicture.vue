<template>
  <div class="wrapper-img">
    <img
      :class="showHero ? 'show-hero' : 'hidden-hero'"
      :src="imgSrc"
      alt="hero"
    />
  </div>
</template>

<script>
export default {
  props: {
    image: { type: String, required: true },
    showHero: {
      type: Boolean,
      required: true,
      default: false,
    },
  },
  //   data(){
  //     return this.heroId
  //   },
  computed: {
    imgSrc() {
      return this.image;
    },
  },
};
</script>

<style scoped>
.wrapper-img {
  padding: 2rem;
}

.hidden-hero {
  filter: blur(50px);
}

.show-hero {
  filter: blur(0);
}
</style>

