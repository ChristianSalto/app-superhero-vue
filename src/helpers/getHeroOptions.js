
import { heroApi, getHeroById } from '@/api/heroApi'
import { Promise } from 'core-js'



export const getHeros = () => {

    const herosArr = Array.from(Array(731))

    return herosArr.map((_, index) => index + 1)

}


export const getHeroOptions = async () => {

    const mixedHeros = getHeros().sort(() => Math.random() - 0.5)

    const heros = await getHeroNames(mixedHeros.splice(0, 4))

    return heros;

}


export const getHeroNames = async ([a, b, c, d] = []) => {

    const promiseArr = [
        await getHeroById(`${a}`),
        await getHeroById(`${b}`),
        await getHeroById(`${c}`),
        await getHeroById(`${d}`)
    ]

    const [hero1, hero2, hero3, hero4] = await Promise.all(promiseArr)
    // const res = await getHeroById(`${a}`)

    //const res = heroApi.get(`/${a}`).then(data => console.log(data))
    console.log(hero1)
    
    return [
        { name: hero1.name, id: hero1.id, image: hero1.image.url },
        { name: hero2.name, id: hero2.id, image: hero2.image.url },
        { name: hero3.name, id: hero3.id, image: hero3.image.url },
        { name: hero4.name, id: hero4.id, image: hero4.image.url },
    ]
}

