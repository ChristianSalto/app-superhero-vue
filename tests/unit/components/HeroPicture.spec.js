import { shallowMount } from '@vue/test-utils'
import HeroPicture from '@/components/HeroPicture'
import { getHeroNames } from '@/helpers/getHeroOptions'

describe('HeroPicture component', () => {
    test('debe de hacer match con el snapshot', () => {
        const wrapper = shallowMount(HeroPicture, {
            props: {
                image: "1",
                showHero: false
            }
        });

        // console.log(wrapper);
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('debe de mostrar la imagen oculta y el heroe 100', () => {
        const wrapper = shallowMount(HeroPicture, {
            props: {
                image: "100",
                showHero: false
            }
        })

        const [img] = wrapper.findAll('img');

        expect(img.exists()).toBeTruthy();
        expect(img.classes('hidden-hero')).toBe(true);

    })

    test('debe de mostrar el hero si esta en true', async () => {
        const wrapper = shallowMount(HeroPicture, {
            props: {
                image: "https://www.superherodb.com/pictures2/portraits/10/100/1.jpg",
                showHero: true
            }
        })

        const [img] = wrapper.findAll('img');

        expect(img.exists()).toBeTruthy();
        expect(img.classes('hidden-hero')).toBe(false);

        const [hero1,hero2,hero3,hero4] = await getHeroNames([1,2,3,4])

        expect(img.attributes('src')).toBe(hero4.image);

        console.log(img.attributes('src'));
    })
})