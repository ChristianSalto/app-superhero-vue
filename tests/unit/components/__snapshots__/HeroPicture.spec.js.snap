// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroPicture component debe de hacer match con el snapshot 1`] = `<div class="wrapper-img"><img class="hidden-hero" src="1" alt="hero"></div>`;
